lib/timers.js

```diff
@@ -481,6 +481,7 @@ Timeout.prototype.unref = function() {
     }
 
     var handle = reuse(this);
+    delete handle._list;
```

Comment:
Under https://github.com/nodejs/node/pull/8372 this code path is far less likely to be hit in the first place.

I doubt the overhead from a later-than-this-tick `unref()` is much of a worry.